integrationtests/tests/tests.rs

```diff
@@ -188,10 +191,133 @@ async fn lightning_gateway_pays_invoice() {
     gateway.server.await_contract_claimed(contract_id).await;
     user.client.fetch_all_coins().await.unwrap();
     assert_eq!(user.total_coins(), sats(2000 - 1010));
-    assert_eq!(gateway.user_client.coins().amount(), sats(1010));
+    assert_eq!(gateway.user.client.coins().amount(), sats(1010));
     assert_eq!(lightning.amount_sent(), sats(1000));
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn receive_lightning_payment_valid_preimage() {
+    let amount = sats(2000);
+    let (fed, user, _, gateway, _) = fixtures(2, 0, &[sats(10), sats(1000)]).await;
+    fed.mint_coins_for_user(&gateway.user, amount).await;
+    assert_eq!(user.total_coins(), sats(0));
+    assert_eq!(gateway.user.total_coins(), amount);
+
+    // Create invoice and offer in the federation
+    let (keypair, invoice) = user
+        .client
+        .create_invoice_and_offer(amount, rng())
+        .await
+        .unwrap();
+
+    // Wait 2 epochs for the offer to achieve consensus
+    fed.run_consensus_epochs(2).await;
+
+    // Gateway deposits ecash to trigger preimage decryption by the federation
+    let (txid, contract_id) = gateway
+        .server
+        .buy_preimage_offer(invoice.payment_hash(), &amount, rng())
+        .await
+        .unwrap();
+    fed.run_consensus_epochs(5).await;
```

Comment:
Nit: would be nice to note why it takes 5 epochs.

Reply:
Done